### PR TITLE
Add lenient option for MoshiConverterFactory

### LINKED ISSUE #1592: MoshiConverterFactory should have a lenient option
https://github.com/square/moshi/issues/132

### PR DESCRIPTION
**Pull request by NightlyNexus**
Closes #1592.

### CODE REVIEW

**Comment by NightlyNexus** [comment]
Is this what you had in mind for #1592?

File: retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiConverterFactory.java [hunk 1]
```diff
@@ -40,20 +40,36 @@ public static MoshiConverterFactory create() {
 
   /** Create an instance using {@code moshi} for conversion. */
   public static MoshiConverterFactory create(Moshi moshi) {
-    return new MoshiConverterFactory(moshi);
+    return create(moshi, false);
+  }
+
+  private static MoshiConverterFactory create(Moshi moshi, boolean lenient) {
```
Comment (JakeWharton) [inline retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiConverterFactory.java]: This doesn't need to exist. Just call the constructor.

File: retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiConverterFactory.java [hunk 2]
```diff
@@ -40,20 +40,36 @@ public static MoshiConverterFactory create() {
...
 
-  private MoshiConverterFactory(Moshi moshi) {
+  private MoshiConverterFactory(Moshi moshi, boolean lenient) {
     if (moshi == null) throw new NullPointerException("moshi == null");
     this.moshi = moshi;
+    this.lenient = lenient;
+  }
+
+  public MoshiConverterFactory asLenient() {
+    if (lenient) {
```
Comment (JakeWharton) [inline retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiConverterFactory.java]: I would just delete this `if`. It'll never happen and even if it does it's not that big of a deal to create a new instance.

**Comment by JakeWharton** [comment]
LGTM otherwise

**Comment by JakeWharton** [comment]
Merged as fb94144db40f8d403ba57799b61e9547cb349b34. Thanks.

**Pull request closed without merging**